### DISCUSSION TITLE
Unable to pass to Docker Run params for maxmemory/maxmemory-policy

### DISCUSSION THREAD

**Discussion by muntdan** [valkey-io/valkey#480] [Q&A]
Hi,

With Redis docker you could send certain params in the RUN command:
docker run -e REDIS_ARGS='--maxmemory 4gb --maxmemory-policy allkeys-lfu'  -p 6379:6379 redis/redis-stack:latest

I attempted the same on the new valkey with both REDIS_ARGS and VALKEY_ARGS
docker run -e REDIS_ARGS='--maxmemory 4gb --maxmemory-policy allkeys-lfu'  -p 6379:6379 valkey/valkey:7.2.5
But when I run CLI  config get maxmemory or config get maxmemory-policy they are 0 and noeviction

How can I set them persistently (not by config set as they are removed after docker restart) on the new valkey docker ?

**Comment by roshkhatri** [comment]
Hi, Valkey docker image would not be a drop in for `redis-stack` but it is a drop in for `redis`.

Can you please try adding your configurations using this section -
### Additionally, If you want to use your own valkey.conf ...
from here: https://hub.docker.com/r/valkey/valkey

**Reply by madolson** [reply]
Maybe we should move this as an issue to the container repo. It might be nice to be able to pass in additional optional arguments to Valkey at startup instead of having to configure the conf file. Something like `VALKEY_ARGS`.

**Reply by muntdan** [reply]
That would be great !

**Reply by madolson** [reply]
Seems like it was already an ask, https://github.com/valkey-io/valkey-container/issues/18. @roshkhatri Do you want to look into that?

**Comment by muntdan** [comment]
Also @roshkhatri to me the documentation option doesnt work if I want to still have the data volume also attached (again a thing that works on redis). If I attempt both volumes the docker starts but is unreachable.

Here are the steps to reproduce:

docker pull valkey/valkey:7.2.5
docker volume create valkey-data
docker volume inspect valkey-data
 
mkdir /var/lib/docker/volumes/valkey-data/conf
wget -P /var/lib/docker/volumes/valkey-data/conf/ https://raw.githubusercontent.com/valkey-io/valkey/7.2.5/valkey.conf
sed -i 's/# maxmemory <bytes>/maxmemory 4GB/g' /var/lib/docker/volumes/valkey-data/conf/valkey.conf
sed -i 's/# maxmemory-policy noeviction/maxmemory-policy allkeys-lfu/g' /var/lib/docker/volumes/valkey-data/conf/valkey.conf

docker run -d -v /valkey-data/:/data -v /var/lib/docker/volumes/valkey-data/conf:/usr/local/etc/valkey --name valkey -p 6379:6379 --memory=4g --cpus=8  valkey/valkey:7.2.5 valkey-server /usr/local/etc/valkey/valkey.conf
